apps/grader-host/src/index.js

```diff
@@ -56,32 +52,38 @@ async.series(
       }
       await lifecycle.init();
     },
-    (callback) => {
-      if (!config.useDatabase) return callback(null);
-      var pgConfig = {
+    async () => {
+      const pgConfig = {
         host: config.postgresqlHost,
         database: config.postgresqlDatabase,
         user: config.postgresqlUser,
         password: config.postgresqlPassword,
         max: config.postgresqlPoolSize,
         idleTimeoutMillis: config.postgresqlIdleTimeoutMillis,
       };
+      function idleErrorHandler(err) {
+        globalLogger.error('idle client error', err);
+        Sentry.captureException(err, {
+          level: 'fatal',
+          tags: {
+            // This may have been set by `sql-db.js`. We include this in the
+            // Sentry tags to more easily debug idle client errors.
+            last_query: err?.data?.lastQuery ?? undefined,
+          },
+        });
+        Sentry.close().finally(() => process.exit(1));
```

Comment:
We kill the process on an idle error in both PrairieLearn and the workspace host; do you think there was a valid reason we weren't doing that for the grader host too?

Reply:
I can't see any reason to not do this in the grader host. In fact the grader host is the one that is the most resilient to errors because of the grading job queue so we should definitely be dying here if something is wrong (like an idle timeout).

Reply:
Isn't the grader host called in the same process as the main PL one in dev mode?

Reply:
For better or worse, PL reimplements most of the external grading execution process for local dev 😭